### ISSUE TITLE
State machine demo bug (video attached)

### ISSUE THREAD

**Issue by farooqali** [sporritt/jsPlumb#377]
Firstly, thanks @sporritt for making jsPlumb! I'm new to jsPlumb so please excuse any misuse of the terms. This problem is easiest to describe in a screen capture of the state machine demo:

https://vimeo.com/128577637

1. Drag a source endpoint away from the source element, then drop to it back to its original location.
2. Drag the target element on the other side of the connection (throws the following error on draw)

jsPlumb function failed : TypeError: Cannot read property 'constructor' of null

Also happens on the latest 1.7.6 branch. It looks like on line 8610 of dom.jspPlumb-1.7.6.js, the anchor is null (otherEndpoint.anchor.constructor == _jp.DynamicAnchor).

**Comment by sporritt** [comment]
i actually don't see this on 1.7.6 dev...i do see it on 1.7.5 though.

**Comment by sporritt** [comment]
@farooqali can you tell me which browser you see this in on the 1.7.6 branch? thanks.

**Comment by farooqali** [comment]
Chrome Version 43.0.2357.65 (64-bit) on Mac 10.10.2

FWIW, the following change seems to fix my own app, which is mostly based
on the demo (but I didn't try the demo on 1.7.6):

1. Removing the if statement in sourceChanged (originalId !== newId) seems
to fix the problem
2. Setting both isSource: true and isTarget: true for the makeSource call

This might be a separate issue, but I've also noticed that connections
still sporadically (but occasionally) get deleted as soon as I move a
connection (drag and drop an endpoint). In those situations, I noticed that
the only event triggered is connectionDetached (in successful drops,
both connectionMoved
and connectionCreated are triggered).

Hope that helps,
Farooq
